unit-tests/live/calib/test_calibrations_common.py

```diff
@@ -85,133 +108,433 @@ def calibration_main(host_assistance, occ_calib, json_config, ground_truth):
         # Stop pipeline
         pipeline.stop()
 
+    if return_table:
+        return health[0], new_calib_result
     return health[0]
 
 
+def measure_depth_fill_rate(ctx=None, width=640, height=480, fps=30, frames=10, timeout_s=12, enable_emitter=True):
+    """Measure average depth fill rate (non-zero pixel ratio * 100) over a number of frames.
+
+    Args:
+        ctx (rs.context|None): Optional existing context; created if None.
+        width, height, fps: Stream configuration.
+        frames (int): Max frames to sample (may exit early after 5 frames if stable).
+        timeout_s (float): Overall timeout.
+        enable_emitter (bool): Enable emitter if supported for consistent fill.
+
+    Returns:
+        float | None: Average fill rate percentage (0-100) or None if capture failed.
+    """
+    try:
+        if ctx is None:
+            ctx = rs.context()
+        cfg = rs.config()
+        cfg.enable_stream(rs.stream.depth, width, height, rs.format.z16, fps)
+        pipe = rs.pipeline(ctx)
```

Comment:
You can set pipeline device with the calibration device you have to reduce time.
Or pass an existing pipeline returned from get_calibration_device